duckduckgo/Android
# Feature/improved bookmarks

Asana Issue URL: https://app.asana.com/0/414730916066338/524278016469964

## Description
Improved bookmark experience

1. Provide the user with a dialog when adding a bookmark (like ios) to allow them to edit the title or url before saving
1. Add "No bookmarks yet" text for empty bookmark pages
1. Show bookmark's domain from the URL in bookmark list
1. Add an overflow button and popup menu with edit and delete options for each bookmark
1. Allow the user to edit the bookmark (use the same dialog as save)


## Steps to Test this PR:
1. Add a new bookmark - ensure you see a dialog prompting for title and URL (both pre-populated)
1. Test adding the bookmark with both the pre-populated values, and edited values
1. Visit the bookmarks activity, verify the new bookmarks are shown, and include the title and domain
1. Use the overflow menu to edit a bookmark; try changing it's title and URL and ensure the bookmarks UI immediately reflects the change
1. Click on the bookmark and ensure it launches the new, edited URL
1. Use the overflow menu to delete a bookmark; ensure it disappears immediately from the activity
1. Delete all bookmarks - ensure there is text shown for this empty bookmark case
